src/instructlab/model/test.py

```diff
@@ -30,54 +36,158 @@
+        test_data_dir = f"{data_dir}/test.jsonl"
+        if not os.path.exists(test_data_dir):
+            click.secho(
+                f"'{test_data_dir}' not such file or directory. Did you run 'ilab model train'?",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        with open(test_data_dir, "r", encoding="utf-8") as f:
+            test_data = [json.loads(line) for line in f]
+
+        print("system prompt:", utils.get_sysprompt())
+        for idx, example in enumerate(test_data):
+            system = example["system"]
+            user = example["user"]
+            print("[{}]\n user prompt: {}".format(idx + 1, user))
+            prompt = f"<|system|>\n{system}\n<|user|>\n{user}\n<|assistant|>"
+            print("expected output:", example["assistant"])
+
+            print("\n-----model output BEFORE training----:\n")
             load_and_train(
-                model=model_dir,
-                adapter_file=adapter_file,
-                max_tokens=100,
-                prompt=prompt,
+                model=model_dir, no_adapter=True, max_tokens=100, prompt=prompt
+            )
+
+            if adapter_file_exists:
+                print("\n-----model output AFTER training----:\n")
+                load_and_train(
+                    model=model_dir,
+                    adapter_file=adapter_file,
+                    max_tokens=100,
+                    prompt=prompt,
+                )
+    else:
+        ctx.obj.logger.debug("")
+        # pylint: disable=import-outside-toplevel
+        # Third Party
+        import httpx
+
+        # First Party
+        from instructlab.model.backends.llama import ensure_server
+
+        # Local
+        from .linux_test import linux_test
+
+        server_process = server_queue = None
+        if not endpoint_url:
+            try:
+                server_process, endpoint_url, server_queue = ensure_server(
+                    ctx.obj.logger,
+                    ctx.obj.config.serve,
+                    True,
+                    "",
+                    "",
+                    "",
+                    model_family,
+                )
+            except Exception as exc:
+                click.secho(f"Failed to start server: {exc}", fg="red")
+                raise click.exceptions.Exit(1)
+        if not endpoint_url:
+            endpoint_url = ctx.obj.config.serve.api_base()
+        ctx.obj.logger.debug("%s", endpoint_url)
+        if not test_file:
+            test_file = sorted(
+                Path(ctx.obj.config.generate.output_dir).glob("test_*"),
+                key=os.path.getmtime,
+                reverse=True,
+            )[0]
+        ctx.obj.logger.debug("test_file=%s", test_file)
+        try:
+            linux_test(
+                ctx,
+                endpoint_url,
+                test_file,
+                api_key=api_key,
+                models=[model, Path("models") / "ggml-model-f16.gguf"],
```

Comment:
seems you are trying to test two models(before and after train),  how did the two models are served? Seems only a default model is served by ensure_server()

Reply:
Good question. Because `linux_train` is still unusable I can't test the full process. Of course I test simulations.

`ensure_server` checks configured model: `"model_path": serve_config.model_path`

I've tested local server which serves local fine-tuned model. Of course it will not work with remote server.

Reply:
You are right `ensure_server` defines `model_path`, `client.chat.completions.create` ignores it.